input_parser/quoting.c

```diff
@@ -0,0 +1,61 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   quoting.c                                          :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: asyed <asyed@student.42.fr>                +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2018/04/03 10:41:08 by satkins           #+#    #+#             */
+/*   Updated: 2018/04/04 20:32:10 by asyed            ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "ast.h"
+
+uint8_t			quoted_flags(char c)
+{
+	if (c == '\'')
+		return (SINGLE_QUOTE);
+	else if (c == '\"')
+		return (DOUBLE_QUOTE);
+	else if (c == '\\')
+		return (BACKSLASH);
+	return (0);
+}
+
+int				handle_embedded_quotes(uint8_t *quoted,
```

Comment:
I'll leave this up to you, but I'd suggest to make this function a distributor function to call other functions depending on the `if/elseif` statements. That way it norms.